src/bin/main.rs

```diff
@@ -163,13 +162,14 @@ async fn main() {
         work_done_progress_params: WorkDoneProgressParams {
             work_done_token: Some(lsp_types::NumberOrString::Number(2)),
         },
+        ..Default::default()
```

Comment:
I am usually not a big fan of defaults, cause it can sometime hide errors in unexpected ways if the default also works for the inner members of the struct

Reply:
This is added because `root_path` is deprecated, but `lsp_types::InitializeParams` doesnt have any other way to construct an instance.
`..Default::default()` is already liberally used in this file.